apps/ledger-live-mobile/src/newArch/storage/index.ts

```diff
@@ -1,34 +1,83 @@
-import type { Storage } from "./types";
+import type { Storage, StorageInitializer, StorageState } from "./types";
 import asyncStorageWrapper from "./asyncStorageWrapper";
+import mmkvStorageWrapper from "./mmkvStorageWrapper";
+import { STORAGE_TYPE } from "./constants";
 
 /** Singleton reference to the global application storage object. */
 export default createStorage();
 
 /** Creates the global application storage object that implements the {@link Storage} interface. */
-function createStorage(): Storage {
+export function createStorage(initializer: StorageInitializer = initStorageState): Storage {
+  const state: StorageState = {
+    storageType: "AsyncStorage",
+  };
+
+  initializer(state);
   return {
     keys() {
-      return asyncStorageWrapper.keys();
+      try {
+        return state.storageType === "MMKV"
```

Comment:
```suggestion
        return state.storageType === STORAGE_TYPE.MMKV
```
This works ? More efficient for safety type